wcfsetup/install/files/lib/system/acp/dashboard/event/AcpDashboardCollecting.class.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace wcf\system\acp\dashboard\event;
+
+use wcf\system\acp\dashboard\box\IAcpDashboardBox;
+use wcf\system\event\IEvent;
+
+/**
+ * Requests the collection of boxes for the acp dashboard.
+ *
+ * @author Marcel Werk
+ * @copyright 2001-2023 WoltLab GmbH
+ * @license GNU Lesser General Public License <http://opensource.org/licenses/lgpl-license.php>
+ * @since 6.1
+ */
+final class AcpDashboardCollecting implements IEvent
+{
+    /**
+     * @var IAcpDashboardBox[]
+     */
+    private array $boxes = [];
+
+    /**
+     * Registers a new box.
+     */
+    public function register(string $name, IAcpDashboardBox $box): void
```

Comment:
I think it would be better if there was `$box->getName()` so that everything about the box is in one place?

Reply:
Agreed. In fact it appears the name is not used at all, except for implicit deduplication.